includes/Discovery.php

```diff
@@ -231,6 +245,56 @@ protected function get_schemaorg_metadata(): array {
 		return apply_filters( 'web_stories_story_schema_metadata', $metadata, $post );
 	}
 
+	/**
+	 * Get product schema data.
+	 *
+	 * @since 1.22.0
+	 *
+	 * @param Story $story Story object.
+	 * @return array
+	 */
+	protected function get_product_data( Story $story ): array {
+		/**
+		 * Product data.
+		 *
+		 * @var array|false $products
+		 */
+		$products = get_post_meta( $story->get_id(), $this->product_meta::PRODUCTS_POST_META_KEY, true );
+		if ( ! \is_array( $products ) ) {
+			return [];
+		}
+		$product_data = [];
+		foreach ( $products as $product ) {
+			$data = [
+				'@type'     => 'Product',
+				'brand'     => $product['productBrand'] ?? '',
+				'productID' => $product['productId'] ?? '',
+				'url'       => $product['productUrl'] ?? '',
+				'name'      => $product['productTitle'] ?? '',
+			];
```

Comment:
Just realized we can also add the product price here.

```suggestion
			$data = [
				'@type'     => 'Product',
				'brand'     => $product['productBrand'] ?? '',
				'productID' => $product['productId'] ?? '',
				'url'       => $product['productUrl'] ?? '',
				'name'      => $product['productTitle'] ?? '',
				'offers'    => [
					[
						'@type'         => 'Offer',
						'price'         => $product['productPrice'] ?? 0,
						'priceCurrency' => $product['productPriceCurrency'] ?? '',
					],
				],
			];
```